### ISSUE TITLE
Confusing log when using LocalCluster(processes=False)

### ISSUE THREAD

**Issue by sami72** [dask/distributed#2084]
When using LocalCluster(processes=False)
the 
`class ThreadPoolExecutor(thread.ThreadPoolExecutor)`
create thread like this 
```
def _adjust_thread_count(self):
            ...
            t = threading.Thread(target=_worker,
                                 name="ThreadPool worker %d" % (len(self._threads,)),


```
This means that we give the same thread name to different threads of different workers
This eventually leads to confusing logs 
when using the following log format 
`%(asctime)s | %(levelname)-5s | %(threadName)s | %(message).10000s'`

**Comment by mrocklin** [comment]
Hrm, interesting.  Is the concern that we might create/destroy/create a thread so that there is more than one thread with the same name in the same process?  Or is the concern that threads from different processes will have the same name?

**Comment by mrocklin** [comment]
Or rather, how do you recommend solving this problem?

**Comment by sami72** [comment]
While running this code 
```
import logging 
import time 
from distributed import Client, LocalCluster
import sys 
import threading 
cluster = LocalCluster(processes=False, n_workers=5)
client = Client(cluster)

logger = logging.getLogger()
logger.setLevel(logging.DEBUG)

ch = logging.StreamHandler(sys.stdout)
ch.setLevel(logging.DEBUG)

formatter = logging.Formatter('%(asctime)s - %(threadName)s - %(levelname)s - %(message)s')
ch.setFormatter(formatter)
logger.addHandler(ch)

def my_task(i):
    print "running task %d in thread %s" % (i, threading.current_thread().ident)
    logger.info("Hello [%d]" , i)
    time.sleep(5)
    logger.info("world [%d]", i)

futures = []
for i in xrange(5):
    futures.append( client.submit(my_task, i+20, pure=False))

client.gather(futures)

```
we get this output 

```
running task 20 in thread 123145468960768
2018-07-02 18:12:18,802 - ThreadPool worker 0 - INFO - Hello [20]
2018-07-02 18:12:18,802 - ThreadPool worker 0 - INFO - Hello [20]
running task 21 in thread 123145473167360
2018-07-02 18:12:18,815 - ThreadPool worker 0 - INFO - Hello [21]
2018-07-02 18:12:18,815 - ThreadPool worker 0 - INFO - Hello [21]
running task 22 in thread 123145477373952
2018-07-02 18:12:18,817 - ThreadPool worker 0 - INFO - Hello [22]
2018-07-02 18:12:18,817 - ThreadPool worker 0 - INFO - Hello [22]
running task 23 in thread 123145481580544
 running task 24 in thread 123145485787136
2018-07-02 18:12:18,819 - ThreadPool worker 0 - INFO - Hello [23]
2018-07-02 18:12:18,819 - ThreadPool worker 0 - INFO - Hello [23]
2018-07-02 18:12:18,819 - ThreadPool worker 0 - INFO - Hello [24]
2018-07-02 18:12:18,819 - ThreadPool worker 0 - INFO - Hello [24]

2018-07-02 18:12:23,809 - ThreadPool worker 0 - INFO - world [20]
2018-07-02 18:12:23,809 - ThreadPool worker 0 - INFO - world [20]
2018-07-02 18:12:23,821 - ThreadPool worker 0 - INFO - world [21]
2018-07-02 18:12:23,821 - ThreadPool worker 0 - INFO - world [21]
2018-07-02 18:12:23,821 - ThreadPool worker 0 - INFO - world [22]
2018-07-02 18:12:23,821 - ThreadPool worker 0 - INFO - world [22]
2018-07-02 18:12:23,824 - ThreadPool worker 0 - INFO - world [24]
2018-07-02 18:12:23,824 - ThreadPool worker 0 - INFO - world [24]
2018-07-02 18:12:23,825 - ThreadPool worker 0 - INFO - world [23]
2018-07-02 18:12:23,825 - ThreadPool worker 0 - INFO - world [23]

```

If you try to read the logs , you can't tell what thread print each line , since the 'threadName' of all the thread is "ThreadPool worker 0"



i suggest that you will do the same thing as done in  _concurrent_futures_thread.py where the 
ThreadPoolExecutor , include the ThreadPoolExecutor counter 
```
self._thread_name_prefix = (thread_name_prefix or
                                    ("ThreadPoolExecutor-%d" % next(self._counter)))
```

**Comment by sami72** [comment]
Hi 
Is that a real problem or just misconfiguration ?
Will it be fixed ?

**Comment by mrocklin** [comment]
It looks like a real problem.  A fix would be quite welcome

**Comment by sami72** [comment]
Great , we will commit a suggestion

**Comment by mrocklin** [comment]
I wonder if maybe we should include the process ID in the name.  That might be both simple and fully specify the thread.

**Comment by sami72** [comment]
Ok

**Comment by mrocklin** [comment]
This was resolved in #2084
